crates/cli/src/dryrun.rs

```diff
@@ -0,0 +1,94 @@
+use core::{
+    chain::Chain,
+    config::Config,
+    encoder::{get_tx_request, try_transactions_from_signal},
+    signals::CrossChainSingleHop,
+};
+use std::{fs, path::PathBuf};
+
+use alloy::signers::local::PrivateKeySigner;
+use color_eyre::eyre::{self, Context as _, ContextCompat};
+use tracing::debug;
+
+#[derive(clap::Args, Debug, Clone)]
+pub(crate) struct DryRun {
+    /// Input signal path
+    #[arg(long)]
+    pub(crate) input_path: PathBuf,
+    /// Output path for the dry run results
+    #[arg(long)]
+    pub(crate) output_path: PathBuf,
+}
+
+impl DryRun {
+    pub(crate) async fn run(&self, config: Config) -> eyre::Result<()> {
+        let data = fs::read_to_string(self.input_path.clone())?;
+        let signal: CrossChainSingleHop =
+            serde_json::from_str(&data).wrap_err("Failed to deserialize signal from input file")?;
+
+        let (slow_signer, fast_signer) = get_signers(
+            &config,
+            signal.slow_chain.clone(),
+            signal.fast_chain.clone(),
+        )?;
+
+        let unsigned_transactions =
+            try_transactions_from_signal(signal.clone(), slow_signer.clone(), fast_signer.clone())
+                .wrap_err("Failed to create transactions from signal")?;
+
+        let slow_tx = unsigned_transactions.0;
+        let fast_tx = unsigned_transactions.1;
+
+        // TODO: get_tx_request should return a signed transaction without broadcasting.
+        let slow_tx_request = get_tx_request(
+            &slow_tx,
+            &slow_signer,
+            &signal.slow_chain.chain_id(),
+            &signal.slow_chain.rpc_url,
+        )
+        .await
+        .wrap_err("Failed to create transaction request for slow chain")?;
+        let fast_tx_request = get_tx_request(
+            &fast_tx,
+            &fast_signer,
+            &signal.fast_chain.chain_id(),
+            &signal.fast_chain.rpc_url,
+        )
+        .await
+        .wrap_err("Failed to create transaction request for fast chain")?;
+
+        let txs = vec![slow_tx_request, fast_tx_request];
+        let txs_json =
+            serde_json::to_string_pretty(&txs).wrap_err("Failed to serialize transactions")?;
+        fs::write(&self.output_path, txs_json).wrap_err_with(|| {
+            format!(
+                "Failed to write slow transaction to file: {}",
+                self.output_path.display()
+            )
+        })?;
+
+        debug!("Transactions written to: {}", self.output_path.display());
+        Ok(())
+    }
+}
+
+fn get_signers(
```

Comment:
just need to fix chain to store the signers instead of the private key string...